public/components/homepage.js

```diff
@@ -0,0 +1,175 @@
+'use strict';
+
+let locationsTemp = [
+  "Bellinzona",
+  "Lugano",
+  "Mendrisio"
+];
+
+class SearchContainer extends React.Component {
+  searchRequest(e) {
+    if (!e.keyCode || e.keyCode == 13) {
+      let searchWarning = document.getElementById('search-warning');
+      let searchName = document.getElementById('search-name').value;
+      if (searchName.length < 3) {
+        searchWarning.innerHTML = "At least 3 characters needed.";
+        return;
+      }
+      if (searchName.match(/[^A-Za-z ]/)) {
+        searchWarning.innerHTML = "Only alphabetic characters allowed.";
```

Comment:
I was asked not to include numbers, so for the moment I'll leave as it is, but I see that it could be useful.